### PR TITLE
Email template notifications

### PR DESCRIPTION
**Pull request by lucferbux**
## Related Issues and Dependencies

[RHODS-852](https://issues.redhat.com/browse/RHODS-852)

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

* Added template folder to create custom mail notification

## Description

Based on the work of [implementing user notifications](https://issues.redhat.com/browse/RHODS-336), we are using [prometheus' built in alert management](https://prometheus.io/docs/alerting/latest/alertmanager/) to trigger email alerts.

## Testing

1. Push your custom odh image as stated in the [testing instructions](https://docs.google.com/document/d/1Ga0khZqr-jGpi5ExPJ0Cz8RawI0INc0UGI6Q2O3fTJc/edit#)
2. Deploy a new version in your dev cluster
3. Inside the `prometheus.yml` file, change [this line](https://github.com/lucferbux/odh-deployer/blob/prometheus_mail_notifications/monitoring/prometheus/prometheus.yaml#L1437) to your redhat email. 
4. Generate a new alert by creating a new Jupyterhub doc and uploading a huge file (more than 2GB)
5. Wait to check the notification. 

## UX Changes

Screenshot 2021-08-25 at 15 41 47

### CODE REVIEW

**Comment by lucferbux** [comment]
@maulikjs and I still need to make some adjustments to the PR.

**Comment by lucferbux** [comment]
@maulikjs @anishasthana @gmfrasca I've updated the description to test the issue, we currently have to figure out one more thing:

- The embedded image inside the email needs a URL just like this one -> https://www.redhat.com/cms/managed-files/Logo-Red_Hat-OpenShift-A-Standard-RGB.png in order to be displayed in gmail. We need a way to store the image in a public URL inside one of our CDNs.
- We should delete or review the [repeated interval](https://github.com/lucferbux/odh-deployer/blob/prometheus_mail_notifications/monitoring/prometheus/prometheus.yaml#L1427), right now is set to 2 mins to test out the feature .

File: monitoring/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -797,6 +1441,16 @@ data:
           alertname: DeadManSnitch
         receiver: deadman-snitch
         repeat_interval: 5m
+      
+      - match:
+          route: user-notifications
+        receiver: user-notifications
+        repeat_interval: 12h
+
+      - match:
```
Comment (maulikjs) [inline monitoring/prometheus/prometheus.yaml:1450]: remove this block of code

Comment (lucferbux) [reply monitoring/prometheus/prometheus.yaml]: 👍

**Review by anishasthana — CHANGES_REQUESTED** [review]
Can you rebase on master and squash commits? Something seems odd about your commits in this PR

**Comment by lucferbux** [comment]
> Can you rebase on master and squash commits? Something seems odd about your commits in this PR
Gonna do that once all changes are commited.

**Comment by lucferbux** [comment]
Ok, so right now there are two things to discuss:

- [Which subject does the emails need](https://github.com/lucferbux/odh-deployer/blob/prometheus_mail_notifications/monitoring/prometheus/prometheus.yaml#L1470)
- [The img tag](https://github.com/lucferbux/odh-deployer/blob/prometheus_mail_notifications/monitoring/prometheus/prometheus.yaml#L1344)

Once we have that sorted out, it will be ready

**Comment by lucferbux** [comment]
> Can you rebase on master and squash commits? Something seems odd about your commits in this PR

Done, rebase from master and squashed all the commits, I'm gonna request your review again

**Review by anishasthana — COMMENTED** [review]
I think this looks good. The only remaining question is about the image

**Comment by lucferbux** [comment]
> I think this looks good. The only remaining question is about the image

Image is not hosted on the repo. Updating the info.

**Comment by anishasthana** [comment]
image

**Review by maulikjs — APPROVED** [review]

**Pull request merged**